File: src/main/java/org/raku/comma/actions/LoadProfileResultsAction.java
package org.raku.comma.actions;

import com.intellij.openapi.actionSystem.ActionUpdateThread;
import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import org.raku.comma.profiler.ProfileResultsChooserDialog;
import org.jetbrains.annotations.NotNull;

public class LoadProfileResultsAction extends AnAction {
    @Override
    public void actionPerformed(@NotNull AnActionEvent e) {
        new ProfileResultsChooserDialog(e.getProject()).show();
    }

    @Override
    public void update(@NotNull AnActionEvent e) {
        e.getPresentation().setEnabled(e.getProject() != null);
    }

    @Override
    public @NotNull ActionUpdateThread getActionUpdateThread() {
        return ActionUpdateThread.BGT;
    }
}

File: src/main/java/org/raku/comma/actions/LoadProfileResultsAction.kt
package org.raku.comma.actions

import com.intellij.openapi.actionSystem.ActionUpdateThread
import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import org.raku.comma.profiler.ProfileResultsChooserDialog

class LoadProfileResultsAction : AnAction() {
    override fun actionPerformed(event: AnActionEvent) {
        ProfileResultsChooserDialog(event.project).show()
    }

    override fun update(event: AnActionEvent) {
        event.presentation.isEnabled = event.project != null
    }

    override fun getActionUpdateThread(): ActionUpdateThread = ActionUpdateThread.BGT
}
